filter_plugins/filters.py

```diff
@@ -70,3 +76,170 @@ def java_arg_build_out(self, java_arg_list):
             if value != '':
                 java_args = java_args + ' ' + value
         return java_args[1:]
+
+    def combine_properties(self, properties_dict):
+        final_dict = {}
+        for prop in properties_dict:
+            if properties_dict[prop].get('enabled'):
+                final_dict.update(properties_dict[prop].get('properties'))
+        return final_dict
```

Comment:
General Comment for this file, can we add comments explaining what the different functions do?

Reply:
Thats a great idea, will do